frontend/rolecall/src/app/performance/performance-editor.component.ts

```diff
@@ -152,10 +152,12 @@ export class PerformanceEditor implements OnInit, OnDestroy, AfterViewChecked {
     this.initStep3Data();
     this.initStep4();
     this.state.status = PerformanceStatus.DRAFT;
-    // Saving casts results in an immediate cast load.
-    // Clear a key parameter to prevent deadly embrace
-    this.castDnD.castSelected = false;
-
+    if (this.castDnD) {
```

Comment:
Oof... interacting directly with ViewChild properties and methods... that's a pretty nasty design pattern. Not sure we'll be able to untangle this.

Reply:
The problem is that this routine, onSaveDraft(), in the next statement, removes the uuid that this.castDnd.selectedCastUUID is pointing at. (I don't want to change this.castDnD, I only want to undo the damage onSaveDraft is already doing). Basically when cast-drag-and-drop get control back after the save, it is going looking for selectedCastUUID and misbehaves when it can't find it.

I agree that it is a dirty coupling. The better way to solve this would probably be to rewrite the cast saving routine so that it somehow clears selectedCastUUID in the cast_api routines (that should be closer to cast-drag-and-drop than the performance-editor) but I don't understand the code on the right level to do that now.